webapp/channels/src/components/channel_header_menu/channel_header_menu_items/channel_header_mobile_menu.tsx

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {ReactNode} from 'react';
+import React, {memo} from 'react';
+
+import type {Channel} from '@mattermost/types/channels';
+
+import MobileChannelHeaderPlugins from '../menu_items/mobile_channel_header_plugins';
+
+type Props = {
+    isMobile: boolean;
+    channel: Channel;
+    pluginItems: ReactNode[];
+}
+
+const ChannelHeaderMobileMenu = (props: Props): JSX.Element => {
+    return (
+        <>
+            {props.isMobile && props.pluginItems}
```

Comment:
Is it correct to render the `pluginItems` here? It looks like they should already be rendered by the various other menu components above using a `MenuItemPluginItems` component, so it seems like they'd be duplicated here

Reply:
Actually, it is. I didn't have any direction on the mobile changes, so I tried to match to existing menu, which flattens and adds Playbooks, if exists, etc. That said, I did have a couple issues. (I have the mobile menus backwards. And I was displaying it as submenu in groups and direct conversations. See images for comparison from old menu.

![Screenshot 2025-03-12 at 6 02 41 PM](https://github.com/user-attachments/assets/f9ce4172-6a03-43ae-b8a0-a540658d0ad4)
![Screenshot 2025-03-12 at 6 01 40 PM](https://github.com/user-attachments/assets/58112992-14d8-4cdb-971f-ce09a0eeb1b7)
![Screenshot 2025-03-12 at 6 01 02 PM](https://github.com/user-attachments/assets/d898f90c-bb91-40b3-ae5d-62df76165cf0)

Reply:
Ah, I missed that the new menu designs for the desktop view moved plugin actions into a submenu.

Thanks for the comparisons. We probably have a bunch of now-unused CSS for the old menu, but I'm glad to know we can get rid of that